pkg/cloudprovider/providers/gce/gce_util.go

```diff
@@ -100,3 +100,25 @@ func isHTTPErrorCode(err error, code int) bool {
 	apiErr, ok := err.(*googleapi.Error)
 	return ok && apiErr.Code == code
 }
+
+// splitProviderID splits a provider's id into core components.
+// A providerID is build out of '${ProviderName}://${project-id}/${zone}/${instance-name}'
+// See cloudprovider.GetInstanceProviderID.
+func splitProviderID(providerID string) (project, zone, instance string, err error) {
```

Comment:
Personally I use regexp only for non trivial things to parse. But a simple regex could easily enhance the readability of this function.  I'd go with something like this`^gce://(.+?)/(.+?)/(.+)$`.